app/assets/javascript/lexxy.js

```diff
@@ -7125,12 +7305,27 @@ class Clipboard {
     item.getAsString((text) => {
       if (isUrl(text) && this.contents.hasSelectedText()) {
         this.contents.createLinkWithSelectedText(text);
+      } else if (isUrl(text)) {
+        const nodeKey = this.contents.createLink(text);
+        this.#dispatchLinkCreatedEvent(nodeKey, { url: text });
       } else {
         this.#pasteMarkdown(text);
       }
     });
   }
 
+  #dispatchLinkCreatedEvent(nodeKey, payload) {
```

Comment:
It'd probably be worth changing it here too for consistency.

```suggestion
  #dispatchLinkInsertEvent(nodeKey, payload) {
```